### PR TITLE
SDK/GCP - Replaced default_gcp_op with task.apply(use_gcp_secret)

### PR DESCRIPTION
**Pull request by Ark-kun**
This change makes it possible to us GCP secret without needing to inherit from a different op.
It also allows users to chain "external" mods that configure tasks:

    Example:
      from kfp.gcp import use_gcp_secret
      task = (
        train_op(...)
          .set_memory_request('1GB')
          .apply(use_gcp_secret('user-gcp-sa'))
          .set_memory_limit('2GB')
      )

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/430)

### CODE REVIEW

**Comment by IronPan** [comment]
This is great! Thanks Alexey

File: sdk/python/kfp/dsl/_container_op.py [hunk 1]
```diff
@@ -88,6 +88,21 @@ def __init__(self, name: str, image: str, command: str=None, arguments: str=None
...
+  def apply(self, mod_func):
+    """Applies a modifier function to self. The function should return the passed object.
+    This is needed to chain "extention methods" to this class.
+
+    Example:
+      from kfp.gcp import use_gcp_secret
+      task = (
+        train_op(...)
+          .set_memory_request('1GB')
+          .apply(use_gcp_secret('user-gcp-sa'))
```
[the remark this answers is not in the document]
Comment (Ark-kun) [reply sdk/python/kfp/dsl/_container_op.py]: In C# terms, `use_gcp_secret` should be an "extension method", not a proper method.

Our customers will want to create dozens of their own methods like this. E.g. `use_aws_secret`, `use_github_secret` etc. We cannot and should not put all of those methods in `ContainerOp`. These should to be functions/methods in some other classes/packages.

The `.apply` method helps chain these modifiers.

[the remark this answers is not in the document]
Comment (Ark-kun) [reply sdk/python/kfp/dsl/_container_op.py]: I'd really prefer that `ContainerOp` remains generic and does not have proprietary methods that a vendor-specific. This is similar to how Kubernetes itself removed explicit nVidia GPU support in favor of vendor-agnostic scheme. I'd like `ContainerOp` to keep being a mirror of Kubernetes `Container` object (which is also and Argo's `container`).

To improve auto-complete we could add have something like

    TrainOp(1,2,3).use_secret(kfp.clouds.gcp)

or

    TrainOp(1,2,3).apply(kfp.clouds.aws)

**Pull request merged**